db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  # Dummy classes, to make migration possible across version changes
+  class Account < ApplicationRecord
+    has_one :user, inverse_of: :account
+    has_one :notification_policy, inverse_of: :account
+  end
+
+  class User < ApplicationRecord
+    belongs_to :account
+  end
+
+  class NotificationPolicy < ApplicationRecord
+    belongs_to :account
+  end
+
+  def up
+    User.includes(account: :notification_policy).find_each do |user|
+      deserialized_settings = Oj.load(user.attributes_before_type_cast['settings'])
+      policy = user.account.notification_policy || user.account.build_notification_policy
+      requires_new_policy = false
+
+      if deserialized_settings['interactions.must_be_follower']
+        policy.filter_not_followers = true
+        requires_new_policy = true
+      end
+
+      if deserialized_settings['interactions.must_be_following']
+        policy.filter_not_following = true
+        requires_new_policy = true
+      end
+
+      if deserialized_settings['interactions.must_be_following_dm']
+        policy.filter_private_mentions = true
+        requires_new_policy = true
+      end
+
+      policy.save if requires_new_policy && policy.changed?
```

Comment:
I see how it's being used - but does the `requires_new_policy` here capture something that `changed?` doesn't capture?

Reply:
If the policy is newly initialized, then `changed?` is true even if we didn't change anything.

Reply:
Ah I see - if it comes from a pure `NotificationPolicy.new` then `changed?` is false to start with, but once we add the account (either by setting, or via `build_*` as done here) then it's already dirty so we need the more specific check on the fields we care about. Given that, fine as-is.

I guess if we really wanted to kill the local var you could do something with `policy.changed` down here and see if any of the relevant attributes are in there.